typescript/cli/src/context/strategies/chain/MultiChainResolver.ts

```diff
@@ -0,0 +1,249 @@
+        chainMetadata: chainMetadata,
+        message: 'Select chains to relay between',
+        requireNumber: 2,
+      });
+      argv.targets = selectedRelayChains.join(',');
+    }
+
+    return [argv.origin, ...argv.targets];
+  }
+
+  private async resolveOriginDestinationChains(
+    argv: Record<string, any>,
+  ): Promise<ChainName[]> {
+    const { chainMetadata } = argv.context;
+
+    argv.origin =
+      argv.origin ??
+      (await runSingleChainSelectionStep(
+        chainMetadata,
+        'Select the origin chain',
+      ));
+
+    argv.destination =
+      argv.destination ??
+      (await runSingleChainSelectionStep(
+        chainMetadata,
+        'Select the destination chain',
+      ));
+
+    return [argv.origin, argv.destination];
+  }
+
+  private async resolveStrategyChains(
+    argv: Record<string, any>,
+  ): Promise<ChainName[]> {
+    const strategy = await readChainSubmissionStrategyConfig(argv.strategy);
+    return extractChainsFromObj(strategy);
+  }
+
+  private async resolveRelayerChains(
+    argv: Record<string, any>,
+  ): Promise<ChainName[]> {
+    return argv.chains.split(',').map((item: string) => item.trim());
+  }
+
+  private async getWarpRouteConfigChains(
+    configPath: string,
+    skipConfirmation: boolean,
+  ): Promise<ChainName[]> {
+    if (!configPath || !isFile(configPath)) {
+      assert(!skipConfirmation, 'Warp route deployment config is required');
+      configPath = await runFileSelectionStep(
+        './configs',
+        'Warp route deployment config',
+        'warp',
+      );
+    } else {
+      log(`Using warp route deployment config at ${configPath}`);
+    }
+
+    // Alternative to readWarpRouteDeployConfig that doesn't use context for signer and zod validation
+    const warpRouteConfig = (await readYamlOrJson(configPath)) as Record<
+      string,
+      any
+    >;
+
+    const chains = Object.keys(warpRouteConfig) as ChainName[];
+    assert(
+      chains.length !== 0,
+      'No chains found in warp route deployment config',
+    );
+
+    return chains;
+  }
+
+  private async resolveCoreApplyChains(
+    argv: Record<string, any>,
+  ): Promise<ChainName[]> {
+    try {
+      const config = readCoreDeployConfigs(argv.config);
+
+      if (!config?.interchainAccountRouter) {
+        return [argv.chain];
+      }
+
+      const addresses = await argv.context.registry.getChainAddresses(
+        argv.chain,
+      );
+      const coreAddresses = DeployedCoreAddressesSchema.parse(
+        addresses,
+      ) as DeployedCoreAddresses;
+
+      const evmCoreModule = new EvmCoreModule(argv.context.multiProvider, {
+        chain: argv.chain,
+        config,
+        addresses: coreAddresses,
+      });
+
+      const transactions = await evmCoreModule.update(config);
+
+      return Array.from(new Set(transactions.map((tx) => tx.chainId))).map(
+        (chainId) => argv.context.multiProvider.getChainName(chainId),
+      );
```

Comment:
You can simplify this by directly using the keys of the `interchainAccountRouter.remoteIcaRouters` field if defined in the config instead of having to run the `core apply` logic here too, because depending on the on-chain config the parsing can take a while. The keys of the `remoteIcaRouters` field are domainIds that can be provided as input to the `getChainName` method of the multiprovider.

Reply:
This wont work, because there are cases that you want apply a change to have no key on `remoteIcaRouters `, but there are keys on chain.
long story short this test case will fail: 
`should unenroll a remote ICA Router and update the config on all involved chains`